minecode_pipelines/miners/cpan.py

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# purldb is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/purldb for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import gzip
+import requests
+
+from bs4 import BeautifulSoup
+
+
+from packageurl import PackageURL
+
+from minecode_pipelines.utils import get_temp_file
+from minecode_pipelines.pipes import write_data_to_json_file
+
+"""
+Visitors for cpan and cpan-like perl package repositories.
+"""
+
+
+CPAN_REPO = "https://www.cpan.org/"
+CPAN_TYPE = "cpan"
+
+
+def get_cpan_packages(cpan_repo=CPAN_REPO, logger=None):
+    cpan_packages_url = cpan_repo + "modules/02packages.details.txt.gz"
+    local_filename = "cpan_packages.gz"
+
+    response = requests.get(cpan_packages_url, stream=True)
```

Comment:
I would use `fetch_http` (`from scanpipe.pipes.fetch import fetch_http`) to handle downloads

You can do
```
cpan_packages_gz_download = fetch_http(cpan_packages_url)
with gzip.open(cpan_packages_gz_download.path, "rb") as f_in:
    ...
```